tests/unity/unity_support.c

```diff
@@ -414,4 +418,37 @@ static void buf_to_xrgb8888(const uint8_t * buf_in, uint8_t * buf_out, lv_color_
     }
 }
 
+static void create_folders_if_needed(const char * path)
+{
+    char * ptr;
+    char * pathCopy = strdup(path);
+    if(pathCopy == NULL) {
+        perror("Error duplicating path");
+        exit(EXIT_FAILURE);
+    }
+
+    char * token = strtok_r(pathCopy, "/", &ptr);
+    char current_path[1024] = {'\0'}; // Adjust the size as needed
+    struct stat st;
+
+    while(token != NULL && *ptr != '\0') {
```

Comment:
`ptr` could become NULL at the end.

```suggestion
    while(token && ptr && *ptr != '\0') {
```

Reply:
I think `!= NULL` is more clear

Reply:
`if (pointer && pointer->elements)`
This coding style is widely used, for example in lv_indev.c. It's also common in NuttX kernel code.

For check null pointer alone, it's normally specified with != NULL explicitly. For example: `LV_ASSERT(pointer != NULL);`

Reply:
Thanks, updated